### PR TITLE
[WIP] Raft db full

### PR DESCRIPTION
**Pull request by jiayang-zheng**
### What is changed and how it works?

Issue Number: Close #xxx

What's Changed:

```commit-message
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
Please add a release note.

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

If you don't think this PR needs a release note then fill it with None.
If this PR will be picked to release branch, then a release note is probably required.
```

### CODE REVIEW

File: components/server/src/server.rs [hunk 1]
```diff
@@ -1473,14 +1477,47 @@ where
                     .get_engine_size()
                     .expect("get raft engine size");
 
+                let mut raftdb_disk_status = disk::DiskUsage::Normal;
+                if seperate_raft_db {
```
Comment (LykxSassinator) [inline components/server/src/server.rs:1481]: A minor advice:
`seperate` => `separate`.

Comment (jiayang-zheng) [reply components/server/src/server.rs]: Fixed. Thanks.

File: components/tikv_util/src/sys/mod.rs [hunk 2]
```diff
@@ -156,3 +160,41 @@ pub fn cache_size(level: usize) -> Option<u64> {
 pub fn cache_line_size(level: usize) -> Option<u64> {
     read_size_in_cache(level, "coherency_line_size")
 }
+
+pub fn path_in_same_mount_point(path1: &str, path2: &str) -> bool {
+    if path1.is_empty() || path2.is_empty() {
+        return true;
+    }
+    let path1 = PathBuf::from(path1);
+    let path1_mount_point = match get_mount(&path1) {
```
Comment (LykxSassinator) [inline components/tikv_util/src/sys/mod.rs:169]: A more concise code style is recommended:

```Rust
let (path1, path2) = (PathBuf::from(path1), PathBuf::from(path2));
match (get_mount(&path1), get_mount(&path2)) {
    (Ok(Some(a)), Ok(Some(b))) => {
        return a == b;
    }
    (Ok(None), Ok(_)) => {
         warn!(...);
        return false;
    }
    (Ok(_), Ok(None)) => {
         warn!(...);
        return false;
    }
    _ => {
        error(....);
        return false;
    }
}

```

File: cmd/tikv-ctl/Cargo.toml [hunk 3]
```diff
@@ -70,7 +70,7 @@ pd_client = { path = "../../components/pd_client", default-features = false }
 prometheus = { version = "0.13", features = ["nightly"] }
 protobuf = { version = "2.8", features = ["bytes"] }
 raft = { version = "0.7.0", default-features = false, features = ["protobuf-codec"] }
-raft-engine-ctl = { git = "https://github.com/tikv/raft-engine.git" }
+raft-engine-ctl = { git = "https://github.com/jiayang-zheng/raft-engine.git", rev = "9200bdae002a2fa09a56e0753eae211477f03f48" }
```
Comment (LykxSassinator) [inline cmd/tikv-ctl/Cargo.toml:73]: Why do u introduce a local / private dependency path here?

File: components/raft_log_engine/Cargo.toml [hunk 4]
```diff
@@ -14,7 +14,7 @@ num_cpus = "1"
 online_config = { path = "../online_config" }
 protobuf = "2"
 raft = { version = "0.7.0", default-features = false, features = ["protobuf-codec"] }
-raft-engine = { git = "https://github.com/tikv/raft-engine.git", features = ["swap"] }
+raft-engine = { git = "https://github.com/jiayang-zheng/raft-engine.git", rev = "9200bdae002a2fa09a56e0753eae211477f03f48", features = ["swap"] }
```
Comment (LykxSassinator) [inline components/raft_log_engine/Cargo.toml:17]: Same as above.

File: components/tikv_util/src/sys/mod.rs (hunk 2 above)
Comment (jiayang-zheng) [reply components/tikv_util/src/sys/mod.rs]: Good Suggestion, thanks!

File: cmd/tikv-ctl/Cargo.toml (hunk 3 above)
Comment (jiayang-zheng) [reply cmd/tikv-ctl/Cargo.toml:73]: Work in process. I will propose a PR in raft-engine repo after test.

File: components/raft_log_engine/Cargo.toml (hunk 4 above)
Comment (jiayang-zheng) [reply components/raft_log_engine/Cargo.toml:17]: This is WIP, I will change it back after raft-engine repo merge related pr.

**Pull request closed without merging**